enterprise/server/invocation_stat_service/invocation_stat_service.go

```diff
@@ -374,6 +400,264 @@ func (i *InvocationStatService) GetInvocationStatBaseQuery(aggColumn string) str
 	return q
 }
 
+func getTableForMetric(metric *sfpb.Metric) string {
+	switch metric.MetricType.(type) {
+	case *sfpb.Metric_Execution:
+		return "Executions"
+	default:
+		return "Invocations"
+	}
+}
+
+func (i *InvocationStatService) getMetricRange(ctx context.Context, table string, metric string, whereClauseStr string, whereClauseArgs []interface{}) (int64, int64, int64, error) {
+	rangeQuery := fmt.Sprintf("SELECT min(%s) as low, max(%s) as high FROM %s %s", metric, metric, table, whereClauseStr)
+	var rows *sql.Rows
+	rows, err := i.olapdbh.DB(ctx).Raw(rangeQuery, whereClauseArgs...).Rows()
+	if err != nil {
+		return 0, 0, 0, err
+	}
+	if !rows.Next() {
+		return 0, 0, 0, nil
+	}
+	valueRange := struct {
+		Low  int64
+		High int64
+	}{}
+	if err := i.olapdbh.DB(ctx).ScanRows(rows, &valueRange); err != nil {
+		return 0, 0, 0, err
+	}
+
+	// A fun hack: make "High" value 1 higher so that we can put an exclusive
+	// limit on the upper bound of requested ranges. Each bucket is [min, max).
+	width := valueRange.High + 1 - valueRange.Low
+	var step int64
+
+	bucketCount := int64(20)
+	if width <= bucketCount {
+		step = 1
+	} else {
+		step = width / bucketCount
+		if width%bucketCount != 0 {
```

Comment:
The format is a bit wierd here. should there be space around '%'

Reply:
This is what my autoformatter does when I have them just out in the open, I think to convey order of operations or something?  I agree it's ugly.  When I put this in parens, it doesn't "fix" it anymore, and I think it's easier on the eyes anyway, so let's go with that.